### PR TITLE
refactor: capture network error + response time data

### PR DESCRIPTION
**Pull request by audrey-kho**
## Description

For each generate request, we will capture additional RUM data for:
- network errors
- response time

## Related Issue

 
https://jira.corp.adobe.com/browse/SITES-21134

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: web-src/src/components/GenerateButton.js [hunk 1]
```diff
@@ -74,12 +74,22 @@ export function GenerateButton() {
     sampleRUM('genai:prompt:generate', { source: 'GenerateButton#handleGenerate' });
     setGenerationInProgress(true);
     setIsOpenPromptEditor(false);
+    const startTime = Date.now();
     generateResults()
       .catch((error) => {
         ToastQueue.negative(error.message, { timeout: 2000 });
+
+        // capture network error (status 500)
+        if (error.status === 500) {
```
Comment (askayastha22) [inline web-src/src/components/GenerateButton.js:83]: We are capturing several error codes from the backend. Can you refer to this file (https://github.com/adobe/aem-genai-assistant/blob/68f48403b7ca801ddb89444990038ecb63740efb/actions/FirefallClient.js#L26) and rewrite this logic?

**Review by askayastha22 — CHANGES_REQUESTED** [review]

**Comment by vtsaplin** [comment]
Perhaps I misunderstood the request, but I assumed it should not be "RUM" but rather EIM.

File: web-src/src/components/GenerateButton.js [hunk 2]
```diff
@@ -74,12 +74,24 @@ export function GenerateButton() {
...
     setIsOpenPromptEditor(false);
+    const startTime = Date.now();
     generateResults()
       .catch((error) => {
         ToastQueue.negative(error.message, { timeout: 2000 });
+
+        // capture network errors
+        if (error.status > 300) {
+          log('prompt:generate:networkerror', { status: error.status, message: error.message });
+          sampleRUM('genai:prompt:generate:networkerror');
```
Comment (askayastha22) [inline web-src/src/components/GenerateButton.js:85]: Please send the error message to RUM as well.

**Review by askayastha22 — CHANGES_REQUESTED** [review]

**Review by askayastha22 — APPROVED** [review]
LGTM

**Pull request merged**